master/rock.py

```diff
@@ -568,18 +574,17 @@ def StandardSetup(c, name, buildconf_url,
           tests=tests, test_utilities=test_utilities)
     BuildReport(build_factory)
 
-    build_properties = {
+    build_properties.update({
         'parallel_build_level': parallel_build_level,
         'parallel_test_level': parallel_test_level
-    }
-
+    })
     build_properties.update(properties)
     c['builders'].append(
         util.BuilderConfig(name=f"{name}-build",
             workernames=build_workers,
             factory=build_factory,
             properties=build_properties,
-            locks=[cache_import_lock.access('counting')]
+            locks=[cache_import_lock.access('counting')] + builder_locks
```

Comment:
What are you using the extra locks for?

Reply:
I have 16 workers, but only 8 builds can run in parallel (there are two types of workers). This allows to make sure only 8 builds get scheduled at any given time even if there are workers available.